src/app/file-manager/transferables-grid/transferables-grid.component.ts

```diff
@@ -195,7 +196,6 @@ export class TransferablesGridComponent implements OnInit, AfterViewInit {
   }
 
   startDownload() {
```

Comment:
Why is this operation here and not in `FileExplorerComponent`?

Reply:
This is because in FileExplorerComponent are the elements to be download. But in transferables-grid is the actual file list to download.

Reply:
OK, I understand. The models aren't quite correct yet; the list of items being downloaded is really a global collection that the transferables grid displays and observes for changes. When an item is queued for download (by closing the file modal dialog in `FileExplorerComponent.selectionDone()`) it should be added to the collection of items being downloaded. The work to start the download occurs outside this UI code, but this code can observe the collection so it knows to update its view of the data being downloaded.

It's similar to what's done here, but with the current code there's no way to know what elements in `filesDatabase` are already downloading and which ones haven't been started yet.

Reply:
Yes, I think I get you. I think this will be fixed in a specific story. When this changes also will changes some lines in the DownloadManager that take care of avoid the repetition of files to download.